### ISSUE TITLE
Remove haptic subsystem in SDL3?

### ISSUE THREAD

**Issue by icculus** [libsdl-org/SDL#6653]
Do we want to remove this outright, or maybe emulate it with just the standard joystick rumble API under the hood?

**Comment by slouken** [comment]
For sdl2-compat we should emulate using the standard joystick rumble API. The main use of this API is for wheels and flight sticks which we don't support in the joystick rumble API.

**Comment by slouken** [comment]
Actually, maybe we don't emulate in sdl2-compat, since wheels and flight sticks won't work there either.

@cgutman was interested in doing the bridge to the gamepad API, but maybe it's not worth it?

**Comment by icculus** [comment]
Yeah, if it's staying, we'll just pass it through from sdl2-compat to SDL3. If we're removing it, we'll either have it always report failure in sdl2-compat, or pull in just enough code from SDL2 Classic to make it work (a luxury we didn't have with SDL 1.2, since the LGPL would forbid it, but something we should avoid in most cases anyhow).

If it's staying, we have a different TODO item for this: right before SDL 2.0.0, we chopped out a haptic effect (Triangle wave or something...I think we took it out to make room for left/right motor support?) because it didn't fit in a Uint16. If we're keeping haptic, we should expand that value to a Uint32 so we can add it back in. I'm hazy on the details at this point, but I can dig them up.
